pkg/raft/raft.go

```diff
@@ -663,13 +671,18 @@ func (r *raft) maybeSendSnapshot(to uint64, pr *tracker.Progress) bool {
 // sendHeartbeat sends a heartbeat RPC to the given peer.
 func (r *raft) sendHeartbeat(to uint64) {
 	pr := r.trk.Progress[to]
+	// If all the cluster nodes are on the new version
+	// then we can safely use leader's commit index in heartbeat.
+	commit := r.raftLog.committed
 	// Attach the commit as min(to.matched, r.committed).
 	// When the leader sends out heartbeat message,
 	// the receiver(follower) might not be matched with the leader
 	// or it might not have all the committed entries.
 	// The leader MUST NOT forward the follower's commit to
 	// an unmatched index.
-	commit := min(pr.Match, r.raftLog.committed)
+	if r.crdbVersion != nil && !r.crdbVersion.IsActive(context.Background(), clusterversion.V24_2_DisableRaftCommitIndexCapping) {
```

Comment:
Do we want to find a way to test this? Maybe we add a mixed-version variant of `lagging_commit.txt`?

Reply:
I am trying to see what is the best way to add the mixed version tests
I am thinking adding parsing logic to assign raft's crdb cluster version handle base on input at
`pkg/raft/rafttest/interaction_env_handler.go`

am i in the ball park? cc @pav-kv

Reply:
Yeah, you could extend the `add-nodes` command with a version parameter. Then, in the test, instead of a single `add-nodes 3` command (which adds 3 nodes configured the same way), you would need to `add-nodes 1` configured one way, and `add-nodes 2` configured the other way.